frontend/src/employee-frontend/components/application-page/ApplicationEditView.tsx

```diff
@@ -220,27 +243,88 @@ export default React.memo(function ApplicationEditView({
               {serviceNeed !== null && (
                 <>
                   <Label>{i18n.application.serviceNeed.partTimeLabel}</Label>
-                  <div>
+                  <FixedSpaceColumn>
                     <Radio
                       label={i18n.application.serviceNeed.partTime}
                       checked={serviceNeed.partTime === true}
                       onChange={() => {
                         setApplication(
                           set('form.preferences.serviceNeed.partTime', true)
                         )
+                        setApplication(
+                          set(
+                            'form.preferences.serviceNeed.serviceNeedOption',
+                            partTimeOptions[0] ?? null
+                          )
+                        )
```

Comment:
Would be nicer to call `setApplication` only once, but since someone has gotten fancy with the lodash functional programming api I'm not 100% sure how. 😅 After reading api docs I **think** the way is 
```
onChange={() => {
   setApplication(
      compose( // add this to lodash/fp imports
         set('form.preferences.serviceNeed.partTime', true),
         set('form.preferences.serviceNeed.serviceNeedOption', partTimeOptions[0] ?? null)
      )
   )
})
```

Reply:
> 
> 
> Would be nicer to call `setApplication` only once, but since someone has gotten fancy with the lodash functional programming api I'm not 100% sure how. 😅 After reading api docs I **think** the way is
> 
> ```
> onChange={() => {
>    setApplication(
>       compose( // add this to lodash/fp imports
>          set('form.preferences.serviceNeed.partTime', true),
>          set('form.preferences.serviceNeed.serviceNeedOption', partTimeOptions[0] ?? null)
>       )
>    )
> })
> ```

I was also thinking that maybe setting the `form.preferences.serviceNeed.serviceNeedOption` to the first value in the list could be done in a `useEffect` that depends on the change of the `form.preferences.serviceNeed.partTime` value.

Reply:
Hmm, I would probably steer away from that because then the update flow gets even more asynchronous. It's not easy to understand when exactly that update would happen and the intermediate state between could cause weird bugs. That was also the point of this whole comment - making the update more atomic and leaving less to be determined by mysterious inner workings of react.

Reply:
I used `flow` instead of `compose`.